rotkehlchen/chain/ethereum/modules/yearn/utils.py

```diff
@@ -0,0 +1,88 @@
+import logging
+from json import JSONDecodeError
+from typing import TYPE_CHECKING
+
+import requests
+
+from rotkehlchen.assets.asset import UnderlyingToken
+from rotkehlchen.assets.utils import get_or_create_evm_token
+from rotkehlchen.chain.ethereum.types import string_to_evm_address
+from rotkehlchen.constants.misc import ONE
+from rotkehlchen.errors.misc import RemoteError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.types import YEARN_VAULTS_V1, YEARN_VAULTS_V2_PROTOCOL, ChainID, EvmTokenKind
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.dbhandler import DBHandler
+
+YEARN_OLD_API = 'https://api.yearn.finance/v1/chains/1/vaults/all'
+
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def query_yearn_vaults(db: 'DBHandler') -> None:
+    """Query yearn API and ensure that all the tokens exists locally. If it exists but the protocol
+    is not the correct one the asset will be edited.
+    TODO: After Optimism gets merged we need to move this logic to a periodic task as we do with
```

Comment:
Yes, I will add the test for the whole task. The reason is that lefteris is touching the same part of the logic where this would go and to avoid conflicts we decided to not plug this logic right now

Reply:
which part is this? I think this can actually go to the ethereum manager now as curve pool is. I agree with Isaac. This needs testing.